datadog/resource_datadog_security_monitoring_rule.go

```diff
@@ -0,0 +1,558 @@
+					},
+					"name": {
+						Type:        schema.TypeString,
+						Optional:    true,
+						Description: "Name of the query.",
+					},
+					"query": {
+						Type:        schema.TypeString,
+						Required:    true,
+						Description: "Query to run on logs.",
+					},
+				},
+			},
+		},
+
+		"tags": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Description: "Tags for generated signals.",
+			Elem:        &schema.Schema{Type: schema.TypeString},
+		},
+	}
+}
+
+func resourceDatadogSecurityMonitoringRuleExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV2 := providerConf.DatadogClientV2
+	authV2 := providerConf.AuthV2
+
+	id := d.Id()
+	_, httpResponse, err := datadogClientV2.SecurityMonitoringApi.GetSecurityMonitoringRule(authV2, id).Execute()
+	if err != nil {
+		if httpResponse != nil && httpResponse.StatusCode == 404 {
+			return false, nil
+		}
+		return false, translateClientError(err, "error checking security monitoring rule exists")
+	}
+	return true, nil
+}
+
+func resourceDatadogSecurityMonitoringRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV2 := providerConf.DatadogClientV2
+	authV2 := providerConf.AuthV2
+
+	ruleCreate, err := buildCreatePayload(d)
+	if err != nil {
+		return err
+	}
+	response, _, err := datadogClientV2.SecurityMonitoringApi.CreateSecurityMonitoringRule(authV2).Body(ruleCreate).Execute()
+	if err != nil {
+		return translateClientError(err, "error creating security monitoring rule")
+	}
+
+	d.SetId(response.GetId())
+
+	return nil
+}
+
+func buildCreatePayload(d *schema.ResourceData) (datadogV2.SecurityMonitoringRuleCreatePayload, error) {
+	payload := datadogV2.SecurityMonitoringRuleCreatePayload{}
+	cases := d.Get("case").([]interface{})
+	payloadCases := make([]datadogV2.SecurityMonitoringRuleCaseCreate, len(cases))
+
+	for idx, ruleCaseIf := range cases {
+		ruleCase := ruleCaseIf.(map[string]interface{})
+		status := datadogV2.SecurityMonitoringRuleSeverity(ruleCase["status"].(string))
+		structRuleCase := datadogV2.NewSecurityMonitoringRuleCaseCreate(status)
+		if v, ok := ruleCase["name"]; ok {
+			name := v.(string)
+			structRuleCase.SetName(name)
+		}
+		if v, ok := ruleCase["condition"]; ok {
+			condition := v.(string)
+			structRuleCase.SetCondition(condition)
+		}
+		if v, ok := ruleCase["notifications"]; ok {
+			tfNotifications := v.([]interface{})
+			notifications := make([]string, len(tfNotifications))
+			for i, value := range tfNotifications {
+				notifications[i] = value.(string)
+			}
+			structRuleCase.SetNotifications(notifications)
+		}
+		payloadCases[idx] = *structRuleCase
+	}
+	payload.Cases = payloadCases
+
+	_, enabled := d.GetOk("enabled")
+	_, disabled := d.GetOk("disabled")
+
+	if enabled && disabled {
+		return payload, errors.New("rule can not be both enabled and disabled")
+	}
```

Comment:
Can it be both "not enabled" and "not disabled"?

Reply:
In this case, neither value has been defined in the conf and the value defaults to `true`